packages/create-next-app/templates/index.ts

```diff
@@ -209,9 +209,9 @@ export const installTemplate = async ({
   if (tailwind) {
     packageJson.devDependencies = {
       ...packageJson.devDependencies,
-      autoprefixer: '^10',
+      autoprefixer: '^10.0.1',
       postcss: '^8',
-      tailwindcss: '^3',
+      tailwindcss: '^3.3.0',
```

Comment:
Even without this change,

<img width="1840" alt="image" src="https://github.com/vercel/next.js/assets/29931815/357b7687-c4a3-4d86-95a2-8651e2f57954">

`tailwindcss` will be resolved as the latest version compatible with `^3`

Reply:
By default, if you use `npm`, the latest version is installed, but since I use `pnpm`, it reuses versions of libraries that have been installed locally in the past that it considers compatible. If I didn't use tailwindcss 3.3.0 or later in local, I'd end up with an older version installed. Of course, you can work around this by manually using a command like `pnpm up` locally, but that doesn't seem like a very clean way to do it in a project that automatically configures itself.

Additionally, I'm attaching the stackblitz url that reproduces the error.
https://stackblitz.com/edit/stackblitz-starters-d7y3uy?file=package.json

Reply:
Perhaps, to fully reproduce this problem, only versions prior to tailwindcsss 3.3.0 must be installed in the pnpm store.

Reply:
check https://github.com/vercel/next.js/pull/56555#issuecomment-1757432818 it was fixed in `pnpm`

Reply:
> check https://github.com/vercel/next.js/pull/56555#issuecomment-1757432818 it was fixed in `pnpm`

As you pointed out, there's already been some PR around this, but in my opinion, the package.json should have everything compatible with the minimum version. Is there any reason why it shouldn't? Leaving everything to the package manager would make it harder to track down problems if you install other dependencies in the future.